### PR TITLE
Add libraries for DB

### PR DESCRIPTION
**Pull request by Deon**
Adding documentation + some DB libraries for now. 

Will be setting up config for db-migrate + adding the migrations from #5 

CC @icechen1

### CODE REVIEW

File: models/User.js [hunk 1]
```diff
@@ -3,6 +3,35 @@ var crypto = require('crypto');
 var mongoose = require('mongoose');
 var Schema = mongoose.Schema;
 
+var knex = require('knex')({
+  client: 'pq',
```
Comment (Deon) [inline models/User.js]: `pg`

File: models/User.js [hunk 2]
```diff
@@ -3,6 +3,35 @@ var crypto = require('crypto');
...
+  connection: {
+    host : '127.0.0.1',
+    user : 'postgres',
+    password : 'postgres',
+    database : 'myapp_test'
+  }
+});
+
+knex.schema.withSchema('public').createTable('users', function (table) {
```
Comment (Deon) [inline models/User.js]: I thought we were running with `node-db-migrate`?

Comment (icechen1) [inline models/User.js]: Ah I was only looking at knex, have not looked at that one yet... I'll check it out tonight

Comment (Deon) [inline models/User.js]: I'm pretty sure if it's here, it'll generate the query everytime this file is loaded in.

**Pull request merged**